### PR TITLE
[v6r20] Add additional shutdown mechanism

### PR DESCRIPTION
**Pull request by RaoulHC**
Added an additional mechanism, such that when the watchdog asks for wall clock seconds left, the mjf utility will also look for a shutdown time file. This can be set to be created by an acpi script when a vm is told to shutdown by the hypervisor.

BEGINRELEASENOTES
*Core
New: Add additional check in MJF utility to look for a shutdown file located at '/var/run/shutdown_time'
ENDRELEASENOTES

### CODE REVIEW

**Comment by fstagni** [comment]
@Andrew-McNab-UK can you review?

**Comment by fstagni** [comment]
@RaoulHC 2 things:
1. you need to use "autopep8" on the files you touched, e.g.:
(from DIRAC directory)
```
autopep8 --global-config=setup.cfg --in-place file1 file2
```

2. It "works" anyway but *your* branch should not be called `integration`

**Comment by RaoulHC** [comment]
@fstagni Thanks, I'll remember not to call the branch integration in future.

**Comment by Andrew-McNab-UK** [comment]
Looks good to me.

File: WorkloadManagementSystem/JobWrapper/Watchdog.py [hunk 1]
```diff
@@ -363,87 +369,85 @@ def _performChecks( self ):
...
+  def __interpretControlSignal(self, signalDict):
     """This method is called whenever a signal is sent via the result of
        sending a sign of life.
     """
-    self.log.info( 'Received control signal' )
-    if type( signalDict ) == type( {} ):
-      if signalDict.has_key( 'Kill' ):
-        self.log.info( 'Received Kill signal, stopping job via control signal' )
+    self.log.info('Received control signal')
+    if isinstance(signalDict, type({})):
```
Comment (fstagni) [inline WorkloadManagementSystem/JobWrapper/Watchdog.py]: I know you didn't code this line, but can you please change it in 
```
if isinstance(signalDict, dict):
```
?

As it is it looks funny

**Pull request merged**